docs/viz.md

```diff
@@ -0,0 +1,320 @@
+
+![tomviz_start_end_pipeline](img/tomviz_start_end_pipeline.png)
+
+Typical datasets may have following relationships between attributes and sizes.
+
+| Volume          | Voxels        | Size (char) | Size (int) |
+|  :---           |  :---         |    :---     | :---       |
+| 64<sup>3</sup>  | 262,144       | 0.25 MB     | 1 MB       |
+| 128<sup>3</sup> | 2,097,152     | 2.00 MB     | 8 MB       |
+| 256<sup>3</sup> | 16,777,216    | 16.00 MB    | 64 MB      |
+| 512<sup>3</sup> | 134,217,728   | 128.00 MB   | 512 MB     |
+| 103<sup>3</sup> | 1,073,741,824 | 1,024.00 MB | 4096 MB    |
+
+### Load data
+
+In this subsection, loading methods for three data categories, single data file, stack of images and raw data set, are being introduced.
+
+#### Single data file
+
+Loading single data set is straight forward, simply select ```Open Data``` from the ```File``` menu, as indicated in screenshot below.
+
+![tomviz_open_data](img/tomviz_open_data.png)
+
+#### Image stacks
+
+Loading image stacks takes a little bit more efforts than loading single data file. After selecting ```Open Stack``` from the ```File``` menu, check all the images you would like to include in the pop-up window, as indicated in screenshots below.
+
+![tomviz_open_stack](img/tomviz_open_stack.png)
+
+![tomviz_stack_dialog](img/tomviz_stack_dialog.png)
+
+#### Reading a raw file
+
+Users can also choose to read raw files by defining data dimensions, type, endianness, and etc, as indicated below.
+
+![raw_reader](img/raw_reader.png)
+
+### Save results
+
+#### Save data
+
+Users can save the data by either clicking the ```Save Data``` button from ```File``` (as shown below), or simply using the keyboard short-cut ```Ctrl+S```.
+
+![tomviz_save_data](img/tomviz_save_data.png)
+
+#### Save state
+
+Similary to saving data, users can save the state by clicking the ```Save State``` button from ```File```.
+
+![tomviz_save_state](img/tomviz_save_state.png)
+
+### Recover and load state
+
+#### Recover state
+
+Tomviz saves the pipeline every five minutes, users can recover the previous states by simply allowing the Tomviz to load them.
+
+![tomviz_recover](img/tomviz_recover.png)
+
+#### Load state
+
+When there is no prompt, users can manually load and recover previous states by selecting ```Load State``` from ```File```.
+
+![tomviz_load_state](img/tomviz_load_state.png)
+
+## Visualization techniques
+
+In this section, we will go over some available techniques and explain the important parameters. Most of the techniques are GPU accelerated, which requires a good graphics card that has at least 1GB memory.
+
+### Available visualization modules
+
+![viz_modules](img/viz_modules.png)
+
+Visualization modules were implemented and optimized using C++ and GLSL (GLslang) to take full advantage of hardware accelerations. These modules are available from the visulization menu.
```

Comment:
Not sure what `GLslang` in brackets is, suggest removing it. We can link to GLSL in a future iteration if people want to know more about. There are typos such as `visulization` that I don't think were in the original. If using VS Code you can install a spell checker plugin and run it over this file.

Reply:
I was not aware of GLSL, so I searched for it and it said "GLSL (GLslang) is a short term for the official OpenGL Shading Language.", which is why I put GLslang in the brackets. I can remove it though, if the abbreviation itself is better self-explained. 

Thank you for letting me know that VS Code has spell checker plugin, I just installed it and fixed all the typos.

Reply:
I think the abbreviation can just link to a page on GLSL definition, it is not central to the text other than to point out that things are hardware accelerated.